src/wallet/crypter.cpp

```diff
@@ -267,6 +301,58 @@ bool CCryptoKeyStore::GetPubKey(const CKeyID &address, CPubKey& vchPubKeyOut) co
     return false;
 }
 
+bool CCryptoKeyStore::AddSpendingKeyPaymentAddress(const libzcash::SpendingKey &key, const libzcash::PaymentAddress &address)
+{
+    {
+        LOCK(cs_KeyStore);
+        if (!IsCrypted())
+            return CBasicKeyStore::AddSpendingKeyPaymentAddress(key, address);
+
+        if (IsLocked())
+            return false;
+
+        std::vector<unsigned char> vchCryptedSecret;
+        CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+        ss << key;
+        CKeyingMaterial vchSecret(ss.begin(), ss.end());
+        if (!EncryptSecret(vMasterKey, vchSecret, address.GetHash(), vchCryptedSecret))
+            return false;
+
+        if (!AddCryptedSpendingKey(address, vchCryptedSecret))
+            return false;
+    }
+    return true;
+}
+
+bool CCryptoKeyStore::AddCryptedSpendingKey(const libzcash::PaymentAddress &address, const std::vector<unsigned char> &vchCryptedSecret)
```

Comment:
The arguments here and for ``AddSpendingKeyPaymentAddress`` have the ``PaymentAddress`` in different positions.